subprojects/core/src/main/java/org/gradle/api/internal/artifacts/dsl/dependencies/DefaultArtifactTransformTargets.java

```diff
@@ -20,16 +20,22 @@
 import org.gradle.api.artifacts.transform.ArtifactTransformTargets;
 import org.gradle.api.attributes.AttributeContainer;
 import org.gradle.api.internal.attributes.AttributeContainerInternal;
-import org.gradle.api.internal.attributes.DefaultAttributeContainer;
+import org.gradle.api.internal.attributes.DefaultMutableAttributeContainer;
+import org.gradle.api.internal.attributes.ImmutableAttributesFactory;
 
 import java.util.List;
 
 public class DefaultArtifactTransformTargets implements ArtifactTransformTargets {
+    private final ImmutableAttributesFactory cache;
 
```

Comment:
Should we better use `attributesFactory` as field/parameter name consistently throughout the code base? Using `cache` here and there might confuse someone who is new to this. One might get the impressions these are two different concepts. (While it always is a factory which optimises memory usage by using caching.)